vm/chef/cookbooks/postgresql/files/pgcluster-utils

```diff
@@ -0,0 +1,190 @@
+#!/bin/bash -eu
+#
+# Copyright 2021 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+source /opt/c2d/c2d-utils || exit 1
+
+readonly postgres_username="postgres"
+readonly sample_table="countries"
+readonly pghba_path="/etc/postgresql/13/main/pg_hba.conf"
+
+#######################################
+# Setup bucardo and check installation
+#######################################
+function setup_bucardo() {
+  which bucardo > /dev/null
+  if [[ "$?" -ne 0 ]]; then
+    echo "Bucardo not found."
+    exit 1
+  fi
+
+  mkdir -p /var/run/bucardo
+}
+
+#######################################
+# Executes a given command locally using default postgres user.
+# Arguments:
+#   Command to be ran.
+#######################################
+function execute_query() {
+  local -r command="$1"
+
+  if [[ -z "${PGPASSWORD}" ]]; then
+    echo "PGPASSWORD environment variable is not set."
+    exit 1
+  fi
+
+  echo "Executing query: ${command}"
+  psql "postgresql://${postgres_username}:${PGPASSWORD}@localhost" -c "${command}"
+}
+
+#######################################
+# Ensure all given instances have the replication table and bucardo user can access it.
+# All replicas will be tested, but not master one.
+# Arguments:
+#   Database name.
+#   Instances list.
```

Comment:
I saw that `pgsql-nodes` is coming from attributes. Is it user defined or will be generated automatically? 

I think it's worth adding here and above `pgsql-nodes` attribute and in metadata what is desired list format.
`"node1 node2"` or `"node1,node2"` etc.

Reply:
`pg-nodes` come from autogen/jinja generated automatically as `elkstack/autogen.yaml;l=43` solution.